gateway/api/repositories/groups.py

```diff
@@ -0,0 +1,51 @@
+"""
+Repository implementation for Groups model
+"""
+
+from typing import List
+from django.contrib.auth.models import Group, Permission
+from django.db.models import Q
+
+from api.models import RUN_PROGRAM_PERMISSION, VIEW_PROGRAM_PERMISSION
+
+
+class GroupRepository:
```

Comment:
As we commented internally instead of `UserGroup` I see it more related with `User` in general so we can include there the operations related with the user.

Reply:
Done [d50a4d2](https://github.com/Qiskit/qiskit-serverless/pull/1560/commits/d50a4d297d139620a6add82cf3d6a40358cb7567)